ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/transaction/arjunacore/jca/SubordinateTransaction.java

```diff
@@ -113,4 +116,28 @@ public void doOnePhaseCommit () throws IllegalStateException,
     public Xid baseXid();
 
     public Uid get_uid();
+
+    /**
+     * <p>
+     * Listing deferred throwables gathered during the transaction processing.
+     * This list contains exceptions thrown by transaction resources when an error
+     * occurs.
+     * <p>
+     * Depending on the transaction stage the particular method returns only an error code e.g. {@link #doPrepare()}
+     * then the controller of the subordinate transaction throws its own exception and it could
+     * pack these throwables in some details.
+     * Other methods as {@link #doCommit()} throws an exception but those exception does not contain
+     * underlying exceptions like those coming from resource failures.
+     * <p>
+     * Normally exceptions taken by this call are put under suppressed part of the thrown exception.
+     *
+     * <p>
+     * JTS implementation of the {@link SubordinateTransaction} interface does not support the deferred throwable
+     * based on the limitation of ORB protocol. This method called on JTS implementation causes {@link UnsupportedOperationException}
+     * being thrown.
+     *
+     * @return list of throwables causing the failure state of the transaction
+     * @throws UnsupportedOperationException for JTS implementation
+     */
+    public List<Throwable> getDeferredThrowables();
```

Comment:
you could add a supportsDeferredThrowables() boolean to the IFC so that the caller knows what to expect?

This is internal interface, you would need to add it to ImportedTransaction instead I expect.